Marlin/Marlin_main.cpp

```diff
@@ -2383,14 +2370,10 @@ static void clean_up_after_endstop_or_probe_move() {
 
     float measured_z = NAN;
     if (!DEPLOY_PROBE()) {
-      measured_z = run_z_probe(
-        #if HAS_CALIBRATION_PROBE
-          is_calibration
-        #endif
-      ) + zprobe_zoffset;
+      measured_z = run_z_probe(unprotected) + zprobe_zoffset;
 
-      if (raise_after == PROBE_PT_RAISE)
-        do_blocking_move_to_z(current_position[Z_AXIS] + Z_CLEARANCE_BETWEEN_PROBES, MMM_TO_MMS(Z_PROBE_SPEED_FAST));
+      if (raise_after == PROBE_PT_RAISE || unprotected)
+        do_blocking_move_to_z(current_position[Z_AXIS] + (unprotected ? 25 : Z_CLEARANCE_BETWEEN_PROBES), MMM_TO_MMS(Z_PROBE_SPEED_FAST));
```

Comment:
add extreme z_clearance in unprotected mode to deal with extreme dome shapes as well (counterpart of z_probe_low_point override)